frontend/src/upload_file.ts

```diff
@@ -240,9 +262,18 @@ class UploadFile {
     }
 
     upload.start();
-    renderer.addNewUpload(filename, uploadIndex);
+
+    const element = renderer.addNewUpload(filename, uploadIndex);
     this.uploads.push(upload);
     this.uploadStatuses.push("uploading");
+
+    if (this.eventEmitter) {
+      this.eventEmitter.emit("addUpload", {
+        element,
```

Comment:
Did you forgot to pass `metaDataField` here?

Reply:
fixed